internal/testdata/r/terraform_data_resource.tf.tmpl

```diff
@@ -2,7 +2,7 @@
 // changes in resources, not in datasources or outputs.
 // You can pass any [testtemplate.DataCommon] to it.
 
-resource "null_resource" "{{ .RName }}" {
+resource "terraform_data" "{{ .RName }}" {
   triggers = {
```

Comment:
```suggestion
  input = {
```

See https://developer.hashicorp.com/terraform/language/resources/terraform-data#input

This fails the e2e tests currently.